pipeline/config.go

```diff
@@ -50,6 +53,32 @@ func (c Config) BuildOperators(bc operator.BuildContext, defaultOperator operato
 	return operators, nil
 }
 
+func dedeplucateIDs(ops []operator.Config) {
+	typeMap := make(map[string]int)
+	for _, op := range ops {
+		if op.Type() != op.ID() {
+			continue
+		}
+
+		if typeMap[op.Type()] == 0 {
```

Comment:
A nuance here, which I think is slightly less than idea, but probably ok:

I think this will "steal" a valid ID from an operator that hasn't been covered yet.
```
- copy
- copy
- copy1
```
would ideally become
```
- copy
- copy2
- copy1
```
but I believe it will actually be
```
- copy
- copy1
- copy2
```

I suggest we accept this behavior, but we should put a test case in place that ensures it does not change accidentally.

Reply:
the second for loop just below this is for exactly that, it checks all the existing IDs to see if the new ID is already in use and increments the type if already exists. I'll add a test case for it.